### ISSUE TITLE
lib size

### ISSUE THREAD

**Issue by avalanche1** [planttheidea/isit.js#1]
Hi!
Isn't your lib size is too big compared to original is.js?

**Comment by planttheidea** [comment]
Yes, it is bigger than the original is.js, but that is an outcome of splitting out all of the functions into individual files for standalone import access. I'll explain below, but first a comparison of the gzipped minified file (how the browser receives it) size:
- is.js: 4.7kb
- isit.js: 8.0 kb

In the original is.js, the all / any / not functions were all built via loop, and by having everything inside of a single closure then the variable declarations could be held to a minimum due to scope inheritance. In order to facilitate importing the functions individually, I needed to create separate files for every function, which are then compiled via Webpack into the final files in the dist/ folder. 

Something like this:

```
import isitBoolean from "isitjs/lib/boolean";
```

Is impossible with is.js (in that you only import the boolean function instead of the entire library), however a use case like that is much more common for someone using Webpack to compile their project than someone downloading the file and including it as a script tag, and that architecture was even discussed [in one of the issues on is.js](https://github.com/arasatasaygin/is.js/issues/10). In that scenario, you would only bring in what you need versus the entire library, which can easily result in much smaller final project size than if you brought in the entire library to use one or two functions. In the example above, let's say you only use isitBoolean ... now isit.js only adds 301 bytes to your project, instead of 4.7kb.

Bottom line, yes it is larger, in fact almost twice the size after minification and gzipping. There are benefits to that approach, but if the benefits of a 3.3kb smaller file outweigh those benefits for your project then I'd stick with is.js.

_EDIT_

Something I forgot to mention is that this architecture also leans itself to a fairly easy custom build of the library. If you want a custom build, you can just clone the repo, delete the stuff you dont need from src/ and run

```
npm run dist
```

This will build a custom build on your local machine that is targeted to your project's exact needs, and is likely to be much smaller than anything else.
